File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int s: scoville){
            pq.offer(s);
        }
        
        int count = 0;
        while(pq.size() >= 2 && pq.peek() < K){
            int first = pq.poll();
            int second = pq.poll();
            
            int mixed = first + (second * 2);
            pq.offer(mixed);
            count++;
        }
        
        return (!pq.isEmpty() && pq.peek() >= K) ? count : -1;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq as hq

def solution(scoville, K):
    answer = 0
    
    hq.heapify(scoville)

    while True:
        first = hq.heappop(scoville)
        
        if first >= K:
            break
        if first < K and len(scoville) <= 0:
            answer = -1
            break
            
        second = hq.heappop(scoville)
        hq.heappush(scoville, first + (second * 2))
        answer+=1
    
    return answer